csrc/adam/cpu_adam.cpp

```diff
@@ -23,7 +23,8 @@ void Adam_Optimizer::Step(float* _params,
                           float* _exp_avg,
                           float* _exp_avg_sq,
                           size_t _param_size,
-                          __half* dev_params)
+                          __half* dev_params,
+                          bool half_precision)

```

Comment:
@awan-10, @jeffra, @RezaYazdaniAminabadi, sorry i realize this is almost 3 years old, but I will like to understand the introduction of `half_precision` in this PR. Who can I talk to you? Thanks!

For context, this affects a current PR under review 
https://github.com/microsoft/DeepSpeed/pull/5409

Reply:
Hi @tjruwase

I can chat with you on this.
I think this is mostly added here to make sure the right AVX operation is selected for FP32 vs FP16. However, as i see it is now templated in this new PR.
Thanks,
Reza

Reply:
@RezaYazdaniAminabadi, thanks for the response. Your explanation was my guess as well. I think the entire code can be greatly simplified with the template usage and also improved type support in torch. Can you please help to review the new PR and also engage in the conversation there?